### PR TITLE
added fixes for removing fault sequence when delete API, read tenant-…

### PR DESCRIPTION
**Pull request by shaniR**
…config when display multiple versions/multiple states, improved logs

### Proposed changes in this pull request

[List all changes you want to add here. If you fixed an issue, please
add a reference to that issue as well.]

-

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 1]
```diff
@@ -3620,27 +3620,95 @@ public static boolean isLoadAPIContextsAtStartup() {
...
+            APIManagerConfiguration config = ServiceReferenceHolder.getInstance().
+                    getAPIManagerConfigurationService().getAPIManagerConfiguration();
 
-        String displayMultiVersions = config.getFirstProperty(APIConstants.API_STORE_DISPLAY_MULTIPLE_VERSIONS);
-        if (displayMultiVersions == null) {
-            log.warn("The configurations related to show multiple versions of API in APIStore " +
-                    "are missing in api-manager.xml.");
-            return false;
+            displayMultiVersions = config.getFirstProperty(APIConstants.API_STORE_DISPLAY_MULTIPLE_VERSIONS);
+            System.out.println("Getting frtom super tenant config = " + displayMultiVersions);
```
Comment (prasa7) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: Is there any special reason to use System.out.println() ? 
There is a typing error "frtom"

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 2]
```diff
@@ -3620,27 +3620,95 @@ public static boolean isLoadAPIContextsAtStartup() {
...
+            } catch (UserStoreException e) {
+                handleException("UserStoreException thrown when tenant-config.json", e);
+            } catch (RegistryException e) {
+                handleException("RegistryException thrown when getting tenant-config.json", e);
+            } catch (ParseException e) {
+                handleException("ParseException thrown when parsing the tenant-config.json content", e);
+            }
+         }
+        if(!configFileHasKey){
+            System.out.println();
```
Comment (prasa7) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: Is there any special reason to use System.out.println() ?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 3]
```diff
@@ -3620,27 +3620,93 @@ public static boolean isLoadAPIContextsAtStartup() {
...
-        String displayAllAPIs = config.getFirstProperty(APIConstants.API_STORE_DISPLAY_ALL_APIS);
-        if (displayAllAPIs == null) {
-            log.warn("The configurations related to show deprecated APIs in APIStore " +
-                    "are missing in api-manager.xml.");
-            return false;
+    public static boolean isAllowDisplayAPIsWithMultipleStatus() throws APIManagementException{
+        String displayAllAPIs = "false";
+        Boolean configFileHasKey = false;
+        String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain(true);
+        if (tenantDomain != null && !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain))
```
Comment (isharac) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: Is the tenant domain case sensitive?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 4]
```diff
@@ -3620,27 +3620,93 @@ public static boolean isLoadAPIContextsAtStartup() {
...
+                handleException("RegistryException thrown when getting tenant-config.json", e);
+            } catch (ParseException e) {
+                handleException("ParseException thrown when parsing the tenant-config.json content", e);
+            }
+        }
+        if(!configFileHasKey){
+            APIManagerConfiguration config = ServiceReferenceHolder.getInstance().
+                    getAPIManagerConfigurationService().getAPIManagerConfiguration();
+            displayAllAPIs = config.getFirstProperty(APIConstants.API_STORE_DISPLAY_ALL_APIS);
+            if (displayAllAPIs == null) {
```
Comment (isharac) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: What if displayAllAPIs is an empty string?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 5]
```diff
@@ -3620,27 +3620,93 @@ public static boolean isLoadAPIContextsAtStartup() {
...
+                    getAPIManagerConfigurationService().getAPIManagerConfiguration();
 
-        String displayMultiVersions = config.getFirstProperty(APIConstants.API_STORE_DISPLAY_MULTIPLE_VERSIONS);
-        if (displayMultiVersions == null) {
-            log.warn("The configurations related to show multiple versions of API in APIStore " +
-                    "are missing in api-manager.xml.");
-            return false;
+            displayMultiVersions = config.getFirstProperty(APIConstants.API_STORE_DISPLAY_MULTIPLE_VERSIONS);
+
+            if (displayMultiVersions == null) {
```
Comment (isharac) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: if displayMultiVersions is an empty string?

File: components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java [hunk 6]
```diff
@@ -1629,40 +1629,51 @@ public static NativeObject jsFunction_getAllPaginatedAPIs(Context cx, Scriptable
...
+            if (null != state) {
+                if (state == APIConstants.PUBLISHED && displayAPIsWithMultipleStatus) {
+                    statusList = new String[]{APIConstants.PUBLISHED, APIConstants.DEPRECATED};
+                } else if (state == APIConstants.PUBLISHED) {
+                    statusList = new String[]{APIConstants.PUBLISHED};
+                } else if (state == APIConstants.PROTOTYPED) {
+                    statusList = new String[]{APIConstants.PROTOTYPED};
+                }
+            }
+        }finally {
```
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java]: Need space between "}" and "finally"

**Comment by bhathiya** [comment]
@shaniR Please make requireed changes.

**Comment by shaniR** [comment]
will add to product later.

**Pull request closed without merging**